web-registry/src/components/organisms/BatchBasicsForm.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useState } from 'react';
+import { Formik, Form, Field, FieldArray } from 'formik';
+import { Box, IconButton, Link } from '@mui/material';
+import { useTheme, DefaultTheme as Theme } from '@mui/styles';
+import { string, date, object, AnyObjectSchema } from 'yup';
+
+import { Body } from 'web-components/lib/components/typography';
+import InputLabel from 'web-components/lib/components/inputs/InputLabel';
+import OnBoardingCard from 'web-components/lib/components/cards/OnBoardingCard';
+import TextField from 'web-components/lib/components/inputs/TextField';
+import ControlledTextField from 'web-components/lib/components/inputs/ControlledTextField';
+import { DatePickField } from 'web-components/lib/components/inputs/DatePickField';
+import SaveFooter from 'web-components/lib/components/fixed-footer/SaveFooter';
+import OutlinedButton from 'web-components/lib/components/buttons/OutlinedButton';
+import TrashIcon from 'web-components/lib/components/icons/TrashIcon';
+import { AddCertificationModal } from 'web-components/lib/components/modal/AddCertificationModal';
+import {
+  requiredMessage,
+  invalidDate,
+  vcsRetirementSerialRE,
+  invalidVCSRetirement,
+  invalidVCSID,
+  numericOnlyRE,
+  isValidJSON,
+  invalidJSON,
+} from 'web-components/lib/components/inputs/validation';
+import { NameUrl } from 'web-components/lib/types/rdf';
+import { VCSBatchMetadataLD } from 'web-components/lib/types/rdf/C01-verified-carbon-standard-batch';
+
+import { ProjectSelect, CreditClassSelect } from '../molecules';
+
+export interface BatchBasicsFormValues {
+  classId: string;
+  startDate: Date | null;
+  endDate: Date | null;
+  metadata?: Partial<VCSBatchMetadataLD>;
+  // TODO: using the same metadata object for different field types created bugs,
```

Comment:
could you be more specific? ie which bugs

Reply:
this is a bit annoying, how are we gonna deal with this issue once we'll have multiple credit classes and corresponding metadata fields?